Duplicate error for invalid RequiresNonNull

Take this simple example:

```
import org.checkerframework.checker.nullness.qual.RequiresNonNull;

class ReqNN {
  @RequiresNonNull("foo")
  void bar() {}
}
```

Running the Nullness Checker gives:

```
ReqNN.java:5: error: [flowexpr.parse.error] cannot parse the expression 'foo: identifier not found'
  void bar() {}
       ^
ReqNN.java:5: error: [flowexpr.parse.error] cannot parse the expression 'foo: identifier not found'
  void bar() {}
       ^
2 errors
```

One error would be better.

Running javac with `-doe` shows the two locations where the error is generated:

```
...
	at org.checkerframework.framework.source.SourceChecker.report(SourceChecker.java:1460)
	at org.checkerframework.framework.flow.CFAbstractTransfer.addInformationFromPreconditions(CFAbstractTransfer.java:533)
	at org.checkerframework.framework.flow.CFAbstractTransfer.initialStore(CFAbstractTransfer.java:255)
	at org.checkerframework.framework.flow.CFAbstractTransfer.initialStore(CFAbstractTransfer.java:94)
	at org.checkerframework.dataflow.analysis.Analysis.init(Analysis.java:440)
	at org.checkerframework.dataflow.analysis.Analysis.performAnalysis(Analysis.java:161)
...
```

and

```
...
	at org.checkerframework.framework.source.SourceChecker.report(SourceChecker.java:1460)
	at org.checkerframework.common.basetype.BaseTypeVisitor.checkContractsAtMethodDeclaration(BaseTypeVisitor.java:524)
	at org.checkerframework.common.basetype.BaseTypeVisitor.visitMethod(BaseTypeVisitor.java:458)
	at org.checkerframework.checker.initialization.InitializationVisitor.visitMethod(InitializationVisitor.java:355)
	at org.checkerframework.checker.initialization.InitializationVisitor.visitMethod(InitializationVisitor.java:57)
...
```

--- a/checker/jtreg/nullness/preciseErrorMsg/Class1.java
+++ b/checker/jtreg/nullness/preciseErrorMsg/Class1.java
@@ -1,0 +1,15 @@
+/*
+ * @test
+ * @summary
+ * Test case for Issue 1051
+ * https://github.com/typetools/checker-framework/issues/1051
+ *
+ * @compile/fail/ref=Class1.out -XDrawDiagnostics -Xlint:unchecked -processor org.checkerframework.checker.nullness.NullnessChecker -AprintErrorStack -Anomsgtext Class1.java
+ *
+ */
+import org.checkerframework.checker.nullness.qual.RequiresNonNull;
+
+public class Class1 {
+    @RequiresNonNull("instanceField")
+    public static void foo() {}
+}

--- a/checker/jtreg/nullness/preciseErrorMsg/Class1.out
+++ b/checker/jtreg/nullness/preciseErrorMsg/Class1.out
@@ -1,0 +1,2 @@
+Class1.java:14:24: compiler.err.proc.messager: (flowexpr.parse.error)
+1 error

--- a/checker/tests/nullness/RequiresNonNullTest.java
+++ b/checker/tests/nullness/RequiresNonNullTest.java
@@ -143,4 +143,10 @@ class RequiresNonNullTest {
     void test() {
         o = null;
     }
+
+    // Test case for Issue 1051
+    // https://github.com/typetools/checker-framework/issues/1051
+    //:: error: (flowexpr.parse.error)
+    @RequiresNonNull("thisShouldIssue1Error")
+    void testIssue1051() {}
 }

--- a/checker/tests/nullness/RequiresNonNullTest.java
+++ b/checker/tests/nullness/RequiresNonNullTest.java
@@ -144,9 +144,9 @@ class RequiresNonNullTest {
         o = null;
     }
 
+    @RequiresNonNull("thisShouldIssue1Error")
     // Test case for Issue 1051
     // https://github.com/typetools/checker-framework/issues/1051
     //:: error: (flowexpr.parse.error)
-    @RequiresNonNull("thisShouldIssue1Error")
     void testIssue1051() {}
 }

--- a/framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java
+++ b/framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java
@@ -529,8 +529,7 @@ public abstract class CFAbstractTransfer<
                                 false);
                 info.insertValue(expr, annotation);
             } catch (FlowExpressionParseException e) {
-                // report errors here
-                analysis.checker.report(e.getResult(), methodTree);
+                // Errors are reported by BaseTypeVisitor.checkContractsAtMethodDeclaration()
             }
         }
     }

